geode-core/src/distributedTest/java/org/apache/geode/cache/ConnectionPoolDUnitTest.java

```diff
@@ -2385,544 +1714,384 @@ public void test017ExpireDestroyHasEntryInCallback() throws CacheException {
+      // no interest registration for region 2
 
-        Region rgn = createRegion(name1, factory.create());
-        rgn.registerInterestRegex(".*", false, false);
-        createRegion(name2, factory.create());
-        // no interest registration for region 2
-      }
     });
 
-    SerializableRunnable populate = new CacheSerializableRunnable("Populate region") {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region1 = getRootRegion().getSubregion(name1);
-        for (int i = 0; i < 10; i++) {
-          region1.put(i, "Region1Old" + i);
-        }
-        Region<Object, Object> region2 = getRootRegion().getSubregion(name2);
-        for (int i = 0; i < 10; i++) {
-          region2.put(i, "Region2Old" + i);
-        }
+    Stream.of(vm1, vm2).forEach(vm -> vm.invoke("Populate region", () -> {
+      Region<Object, Object> region1 = getRootRegion().getSubregion(name1);
+      for (int i = 0; i < 10; i++) {
+        region1.put(i, "Region1Old" + i);
       }
-    };
-    vm1.invoke(populate);
-    vm2.invoke(populate);
+      Region<Object, Object> region2 = getRootRegion().getSubregion(name2);
+      for (int i = 0; i < 10; i++) {
+        region2.put(i, "Region2Old" + i);
+      }
+    }));
 
-    vm1.invoke("Update", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region1 = getRootRegion().getSubregion(name1);
-        for (int i = 0; i < 10; i++) {
-          region1.put(i, "Region1New" + i);
-        }
-        Region<Object, Object> region2 = getRootRegion().getSubregion(name2);
-        for (int i = 0; i < 10; i++) {
-          region2.put(i, "Region2New" + i);
-        }
+    vm1.invoke("Update", () -> {
+      Region<Object, Object> region1 = getRootRegion().getSubregion(name1);
+      for (int i = 0; i < 10; i++) {
+        region1.put(i, "Region1New" + i);
+      }
+      Region<Object, Object> region2 = getRootRegion().getSubregion(name2);
+      for (int i = 0; i < 10; i++) {
+        region2.put(i, "Region2New" + i);
       }
     });
 
     // Wait for updates to be propagated
-    Wait.pause(5 * 1000);
-
-    vm2.invoke("Validate", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region1 = getRootRegion().getSubregion(name1);
-        for (int i = 0; i < 10; i++) {
-          assertEquals("Region1New" + i, region1.get(i));
-        }
-        Region<Object, Object> region2 = getRootRegion().getSubregion(name2);
-        for (int i = 0; i < 10; i++) {
-          assertEquals("Region2Old" + i, region2.get(i));
-        }
+    vm2.invoke("Validate", () -> {
+      Region<Object, Object> region1 = getRootRegion().getSubregion(name1);
+      for (int i = 0; i < 10; i++) {
+        final int testInt = i;
+        await().until(() -> ("Region1New" + testInt).compareTo((String) region1.get(testInt)) == 0);
```

Comment:
I recommend rewording this without compareTo, especially for the error message in the assertion form:
```
await().until(() -> ((String) region1.get(testInt)).equals("Region1New" + testInt));
or
await().untilAsserted(() -> assertThat(((String) region1.get(testInt))).isEqualTo("Region1New" + testInt));
```